CMakeLists.txt

```diff
@@ -1,81 +1,472 @@
+
+  # Precision-based Fortran compiler flags
+  set(r8_flags "-fdefault-real-8 -fdefault-double-8") # Fortran flags for 64BIT precision
+
+  # Copied from MOM5/bin/mkmf.template.nci.gfortran
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -ffree-line-length-none -fno-range-check -Waliasing -Wampersand -Warray-bounds -Wcharacter-truncation -Wconversion -Wline-truncation -Wintrinsics-std -Wsurprising -Wno-tabs -Wunderflow -Wunused-parameter -Wintrinsic-shadow -Wno-align-commons")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g -W -fbounds-check")
+
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+
+  # Precision-based Fortran compiler flags
+  set(r4_flags "-real-size 32") # Fortran flags for 32BIT precision
+  set(r8_flags "-real-size 64") # Fortran flags for 64BIT precision
+  set(r8_flags "${r8_flags} -no-prec-div -no-prec-sqrt")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv")
+
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "IntelLLVM")
+
+  # Precision-based Fortran compiler flags
+  set(r4_flags "-real-size 32") # Fortran flags for 32BIT precision
+  set(r8_flags "-real-size 64") # Fortran flags for 64BIT precision
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv")
+
+else()
+  message(WARNING "Fortran compiler with ID ${CMAKE_Fortran_COMPILER_ID} will be used with CMake default options")
 endif()
 
-include_directories(${MPI_Fortran_INCLUDE_PATH})
-add_compile_options(${MPI_Fortran_COMPILE_FLAGS})
+################################################################################
+# C
+################################################################################
+
+if(CMAKE_C_COMPILER_ID MATCHES "GNU")
+
+  # Copied from MOM5/bin/mkmf.template.nci.gfortran
+  set(CMAKE_C_FLAGS         "${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG   "-O0 -g")
+  set(CMAKE_C_FLAGS_RELEASE "-O2")
+
+elseif(CMAKE_C_COMPILER_ID STREQUAL "Intel")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_C_FLAGS         "${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG   "-O0 -g -ftrapuv -traceback")
```

Comment:
Flag mismatch between C and Fortran compilers. Fortran has `-traceback` on for `RELEASE`, which I agree with BTW.

Reply:
In this PR, let's keep the compiler flags identical to MOM5. Diverging from MOM5 compiler flags should be a separate issue.